CMakeLists.txt

```diff
@@ -24,6 +24,8 @@ endif()
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
We could remove "-std=c++11" if we set `set(CMAKE_CXX_STANDARD 11)`

Reply:
Without that flag, the Mac with CUDA 8, MacOS 10.12.6 and Apple llvm 8.0.0 prevents the compiler from opening the file <future>: "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/future(1113): here is inaccessible"

Reply:
Arf, cmake craziness...
Same problem described here: https://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake